pkg/kubelet/images/image_manager.go

```diff
@@ -22,28 +22,50 @@ import (
 
 	dockerref "github.com/docker/distribution/reference"
 	v1 "k8s.io/api/core/v1"
+	utilfeature "k8s.io/apiserver/pkg/util/feature"
 	"k8s.io/client-go/tools/record"
 	"k8s.io/client-go/util/flowcontrol"
 	"k8s.io/klog/v2"
 
 	runtimeapi "k8s.io/cri-api/pkg/apis/runtime/v1"
+	"k8s.io/kubernetes/pkg/credentialprovider"
+	credentialprovidersecrets "k8s.io/kubernetes/pkg/credentialprovider/secrets"
+	"k8s.io/kubernetes/pkg/features"
 	kubecontainer "k8s.io/kubernetes/pkg/kubelet/container"
 	"k8s.io/kubernetes/pkg/kubelet/events"
+	"k8s.io/kubernetes/pkg/util/parsers"
 )
 
+// ensureSecretPulledImage - map of imageref (image digest) to successful secret pulled image details
+var ensureSecretPulledImage map[string]*ensureSecretPulledImageDigest
+
+type ensureSecretPulledImageDigest struct {
+	// TODO: (mikebrow) time of last pull for this imageRef
+	// TODO: (mikebrow) time of pull for each particular auth hash
+
+	// map of auth hash (keys) used to successfully pull this imageref
+	HashMatch map[string]bool
```

Comment:
nit: Could use https://github.com/kubernetes/apimachinery/blob/master/pkg/util/sets/string.go

Reply:
per our conversation adding TODO to line 44 :-)